geowebcache/core/src/main/java/org/geowebcache/config/XMLConfiguration.java

```diff
@@ -135,6 +135,13 @@
     private String templateLocation;
 
     private GridSetBroker gridSetBroker;
+    
+    /**
+     * A flag for whether the config needs to be loaded at {@link #initialize(GridSetBroker)}. If
+     * the constructor loads the configuration, will set it to false, then each call to initialize()
+     * will reset this flag to true
+     */
+    private boolean reloadConfigOnInit = true;
```

Comment:
It's just to avoid reloading the config at initialize(), becasue the xml file gets parsed at construction time now, since other components may need access to certain config objects before initialize() gets called